### PR TITLE
Added Team Webhook

### PR DESCRIPTION
**Pull request by mohitdeuex**
### Describe your changes :

Added Tens Webhook Integratiom

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.openmetadata.catalog.util.ChangeEventParser;
+
+@Slf4j
+public class TeamWebhookEventPublisher extends WebhookPublisher {
+  private final Invocation.Builder target;
+  private final Client client;
+
+  public TeamWebhookEventPublisher(Webhook webhook, CollectionDAO dao) {
+    super(webhook, dao);
+    String slackWebhookURL = webhook.getEndpoint().toString();
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java:26]: lets rename this to msTeamsWebhookURL

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import org.openmetadata.catalog.type.ChangeEvent;
+import org.openmetadata.catalog.type.Webhook;
+import org.openmetadata.catalog.util.ChangeEventParser;
+
+@Slf4j
+public class TeamWebhookEventPublisher extends WebhookPublisher {
+  private final Invocation.Builder target;
+  private final Client client;
+
+  public TeamWebhookEventPublisher(Webhook webhook, CollectionDAO dao) {
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java:24]: also rename this to MSTeamsWebhook, Team itself is hard to identify

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    super(webhook, dao);
+    String slackWebhookURL = webhook.getEndpoint().toString();
+    ClientBuilder clientBuilder = ClientBuilder.newBuilder();
+    clientBuilder.connectTimeout(10, TimeUnit.SECONDS);
+    clientBuilder.readTimeout(12, TimeUnit.SECONDS);
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java:29]: should that connectionTimeout and readTimeout be configurable?

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/util/ChangeEventParser.java [hunk 4]
```diff
@@ -77,7 +78,7 @@ public static String getEntityUrl(ChangeEvent event) {
     if (Objects.nonNull(urlInstance)) {
       String scheme = urlInstance.getScheme();
       String host = urlInstance.getHost();
-      return String.format("<%s://%s/%s/%s|%s>", scheme, host, event.getEntityType(), fqn, fqn);
+      return String.format("<%s://%s/%s/%s>", scheme, host, event.getEntityType(), fqn);
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/util/ChangeEventParser.java:81]: why are we changing this? was it wrong before?

File: catalog-rest-service/src/main/resources/json/schema/entity/events/webhook.json [hunk 5]
```diff
@@ -12,7 +12,7 @@
       "type": "string",
       "javaType": "org.openmetadata.catalog.type.WebhookType",
       "default": "generic",
-      "enum": ["slack", "generic", "kafka"],
+      "enum": ["slack", "generic", "kafka", "teams"],
```
Comment (harshach) [inline catalog-rest-service/src/main/resources/json/schema/entity/events/webhook.json:15]: we need to remove type "Kafka" from here

**Comment by harshach** [comment]
@mohitdeuex left few comments please check

File: openmetadata-ui/src/main/resources/ui/src/utils/SvgUtils.tsx [hunk 6]
```diff
@@ -130,6 +130,7 @@ import Logo from '../assets/svg/logo.svg';
 import IconManageColor from '../assets/svg/manage-color.svg';
 import IconMinus from '../assets/svg/minus.svg';
 import IconMlModal from '../assets/svg/mlmodal.svg';
+import IconMSTeamsGrey from '../assets/svg/ms-teams-grey.svg';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/SvgUtils.tsx:133]: are we using `ms-tems` icon that we added?

**Review by chirag-madlani — APPROVED** [review]

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/util/ChangeEventParser.java (hunk 4 above)
Comment (mohitdeuex) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/util/ChangeEventParser.java]: updated , it needed to be changed for teams

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java (hunk 3 above)
Comment (mohitdeuex) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/events/TeamWebhookEventPublisher.java]: added

File: openmetadata-ui/src/main/resources/ui/src/utils/SvgUtils.tsx (hunk 6 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/utils/SvgUtils.tsx:133]: We were using it just not added her. Updating here also.

**Comment by harshach** [comment]
@mohitdeuex 2 tests are failing on java side. Can you check and fix.

**Comment by mohitdeuex** [comment]
> @mohitdeuex 2 tests are failing on java side. Can you check and fix.

fixed @harshach

**Pull request merged**